src/parallel/DomainDecomposition.cpp

```diff
@@ -53,22 +53,27 @@ void DomainDecomposition::initCommunicationPartners(double cutoffRadius, Domain
 
 void DomainDecomposition::prepareNonBlockingStage(bool /*forceRebalancing*/, ParticleContainer* moleculeContainer,
 		Domain* domain, unsigned int stageNumber) {
-	if(sendLeavingWithCopies()){
-		DomainDecompMPIBase::prepareNonBlockingStageImpl(moleculeContainer, domain, stageNumber, LEAVING_AND_HALO_COPIES);
-	}
-	else {
-		DomainDecompMPIBase::prepareNonBlockingStageImpl(moleculeContainer, domain, stageNumber, LEAVING_ONLY);
-		DomainDecompMPIBase::prepareNonBlockingStageImpl(moleculeContainer, domain, stageNumber, HALO_COPIES);
+	if (sendLeavingWithCopies()) {
+		DomainDecompMPIBase::prepareNonBlockingStageImpl(moleculeContainer, domain, stageNumber,
+														 LEAVING_AND_HALO_COPIES);
+	} else {
+		// Would first need to send leaving, then halo -> not good for overlapping!
+		global_log->error()
+			<< "nonblocking P2P using separate messages for leaving and halo is currently not supported." << std::endl;
```

Comment:
Could you also include info what needs to be changed in the input XML to fix this?